### PR TITLE
Spark

### PR DESCRIPTION
**Pull request by fiatjaf**
This builds on top of https://github.com/ZeusLN/zeus/pull/269 because I'm disorganized.

It abstracts backends into 3 different classes on `RESTUtils.ts` and tries to leave as little as possible of the differences between backends into the rest of the app.

### CODE REVIEW

File: stores/InvoicesStore.ts [hunk 1]
```diff
@@ -229,32 +185,26 @@ export default class InvoicesStore {
...
-                        routes.forEach((route: any) => {
-                            // expect lnd to pick the cheapest route
-                            if (this.feeEstimate) {
-                                if (route.total_fees < this.feeEstimate) {
-                                    this.feeEstimate = route.total_fees;
-                                }
-                            } else {
-                                this.feeEstimate = route.total_fees || 0;
+        return RESTUtils.getRoutes([destination, amount])
+            .then((data: any) => {
```
Comment (kaloudis) [inline stores/InvoicesStore.ts:189]: response: any

File: utils/RESTUtils.ts [hunk 2]
```diff
@@ -111,96 +59,442 @@ class RESTUtils {
...
+                }
+            }
+        });
+    };
+
+    getTransactions = () =>
+        this.rpc('listfunds').then(({ outputs }) => ({
+            transactions: outputs
+        }));
+    getChannels = () =>
```
Comment (kaloudis) [inline utils/RESTUtils.ts:228]: async getChannels() {

File: ios/Podfile [hunk 3]
```diff
@@ -47,12 +46,6 @@ target 'zeus' do
 
   pod 'rn-fetch-blob', :path => '../node_modules/rn-fetch-blob'
 
-  target 'zeusTests' do
```
Comment (kaloudis) [inline ios/Podfile:50]: need to restore this to

```target 'zeusTests' do
        inherit! :search_paths
     end```

It was breaking the iOS build

File: stores/ChannelsStore.ts [hunk 4]
```diff
@@ -170,42 +152,21 @@ export default class ChannelsStore {
                 id: `${request.node_pubkey_string}@${request.host}`
             };
         } else {
-            data = JSON.stringify({
-                addr: {
-                    pubkey: request.node_pubkey_string,
-                    host: request.host
-                }
-            });
+            data = JSON.stringify();
```
Comment (kaloudis) [inline stores/ChannelsStore.ts:155]: what's going on here?

Comment (fiatjaf) [reply stores/ChannelsStore.ts:155]: Oops. Forgot to delete this entire block.

**Comment by fiatjaf** [comment]
Ok, this should be ready to go now.

File: views/Settings/AddEditNode.tsx [hunk 5]
```diff
@@ -423,6 +524,11 @@ export default class AddEditNode extends React.Component<
                                                         'c-lightning-REST',
                                                     saved: false
                                                 });
+                                            } else if (buttonIndex === 2) {
```
Comment (kaloudis) [inline views/Settings/AddEditNode.tsx:527]: `} else if (buttonIndex === **3**) {`

**Pull request merged**